### PR TITLE
Update proto

### PR DESCRIPTION
**Pull request by mradzisz**

### CODE REVIEW

File: proto/tribune_tts.proto [hunk 1]
```diff
@@ -48,13 +60,70 @@ message SynthesizeRequest
 // Provides information to the synthesizer that specifies how to process the request.
 message SynthesizeConfig
 {
-    // Desired sampling frequency in hertz of synthesized audio.
+    // Desired sampling frequency in hertz of synthesized audio. The value 0 means use the default Synthesizer sampling rate.
```
Comment (tomjad) [inline proto/tribune_tts.proto]: This field could be moved to `AudioConfig` message

File: proto/tribune_tts.proto [hunk 2]
```diff
@@ -48,13 +60,70 @@ message SynthesizeRequest
...
     int32 sample_rate_hertz = 1;
+
+    // Requested format of the output audio stream.
+    OutputFormat output_format = 2;
```
Comment (tomjad) [inline proto/tribune_tts.proto]: This field should be renamed to `audio_encoding` and also could be moved to `AudioConfig` message

File: proto/tribune_tts.proto [hunk 3]
```diff
@@ -48,13 +60,70 @@ message SynthesizeRequest
...
+
+    // Optional. Overrides the default synthesized speech parameters
+    AudioConfig audio_config = 4;
+
+    // Requested voice to be used to synthesize the text (may be overriden by SSML tags in request text).
+    Voice voice = 5;
+}
+
+// The requested format of the response audio data.
+enum OutputFormat
```
Comment (tomjad) [inline proto/tribune_tts.proto]: Rename to `AudioEncoding`

File: proto/tribune_tts.proto [hunk 4]
```diff
@@ -48,13 +60,70 @@ message SynthesizeRequest
...
 {
-    // Desired sampling frequency in hertz of synthesized audio.
+    // Desired sampling frequency in hertz of synthesized audio. The value 0 means use the default Synthesizer sampling rate.
     int32 sample_rate_hertz = 1;
+
+    // Requested format of the output audio stream.
+    OutputFormat output_format = 2;
+
+    // ISO 639-1 code of the language of text to be synthesized (may be overriden by SSML tags in request text).
+    string language = 3;
```
Comment (tomjad) [inline proto/tribune_tts.proto]: Does `language` should be placed directly in `SynthesizeConfig`, or `Voice` is better place? Just asking, maybe expected behaviour is that when `voice` is not defined, the default one for selected language is used ?

File: proto/tribune_tts.proto [hunk 5]
```diff
@@ -48,13 +60,70 @@ message SynthesizeRequest
...
+    AudioConfig audio_config = 4;
+
+    // Requested voice to be used to synthesize the text (may be overriden by SSML tags in request text).
+    Voice voice = 5;
+}
+
+// The requested format of the response audio data.
+enum OutputFormat
+{
+    PCM16 = 0;       // Uncompressed 16-bit signed integer samples.
```
Comment (tomjad) [inline proto/tribune_tts.proto]: PCM audio is returned with- or without- the header?

File: proto/tribune_tts.proto [hunk 6]
```diff
@@ -48,13 +60,70 @@ message SynthesizeRequest
...
+    float volume = 4;
+}
+
+// Voice definition. If both name and gender is defined, the name takes precedence.
+// If only name is defined, the specified voice is used.
+// If only gender is defined, the default voice for the requested gender is used.
+message Voice
+{
+    // The name of the voice (empty string means default voice).
+    string name = 1;
```
Comment (tomjad) [inline proto/tribune_tts.proto:101]: add `language_code` here?

File: python/call_synthesize.py [hunk 7]
```diff
@@ -37,8 +50,11 @@ def call_synthesize(args, text):
                 else:
                     ws.setFrameRate(response.audio.sample_rate_hertz)
                 ws.append(response.audio.content)
-                if response.audio.end_of_stream:
-                    ws.save(wavefilename)
+
+        if out_format == tribune_tts_pb2.OutputFormat.PCM16:
```
Comment (tomjad) [inline python/call_synthesize.py]: You can set `out_format` in WaveSaver constructor and here do not use condition

File: python/wave_saver.py [hunk 8]
```diff
@@ -31,7 +31,12 @@ def save(self, filename):
             params = (self._nchannels, self._sampwidth, self._framerate, len(self.buffer), 'NONE', 'not compressed')
             w.setparams(params)
             w.writeframes(self.buffer)
-    
+
+    def saveOggVorbis(self, filename):
```
Comment (tomjad) [inline python/wave_saver.py]: This class should be renamed to `AudioSaver`. If you move passing encoding type to AudioSaver constructor, only one `save` method could be exposed;

**Review by tomjad — CHANGES_REQUESTED** [review]

File: proto/tribune_tts.proto (hunk 1 above)
Comment (mradzisz) [reply proto/tribune_tts.proto]: Moved.

File: proto/tribune_tts.proto (hunk 2 above)
Comment (mradzisz) [reply proto/tribune_tts.proto]: Renamed and moved.

File: proto/tribune_tts.proto (hunk 3 above)
Comment (mradzisz) [reply proto/tribune_tts.proto]: Renamed.

File: proto/tribune_tts.proto (hunk 5 above)
Comment (mradzisz) [reply proto/tribune_tts.proto]: Without the header. Clarified in comment.

File: proto/tribune_tts.proto (hunk 4 above)
Comment (mradzisz) [reply proto/tribune_tts.proto]: In SSML we have separate tags for switching between languages (<lang>) and voices (<voice>), so perhaps placing a language directly in SynthesizeConfig is OK.

File: proto/tribune_tts.proto (hunk 6 above)
Comment (mradzisz) [reply proto/tribune_tts.proto:101]: In SSML we have separate tags for switching between languages () and voices (), so perhaps placing a language directly in SynthesizeConfig is OK.

File: python/call_synthesize.py (hunk 7 above)
Comment (mradzisz) [reply python/call_synthesize.py]: Done.

File: python/wave_saver.py (hunk 8 above)
Comment (mradzisz) [reply python/wave_saver.py]: Done.

**Review by tomjad — APPROVED** [review]

**Pull request merged**